dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumContextProvider.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.webview
+
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.internal.domain.RumContext
+import com.datadog.android.rum.internal.domain.scope.RumApplicationScope
+import com.datadog.android.rum.internal.domain.scope.RumSessionScope
+import com.datadog.android.rum.internal.monitor.DatadogRumMonitor
+
+internal class RumContextProvider {
+
+    fun getRumContext(): RumContext? {
+        val appScope = (GlobalRum.get() as? DatadogRumMonitor)?.rootScope as? RumApplicationScope
+        val rumContext = (appScope?.childScope as? RumSessionScope)?.getRumContext()
+        return if (rumContext == null ||
+            rumContext.applicationId == RumContext.NULL_UUID ||
+            rumContext.sessionId == RumContext.NULL_UUID
+        ) {
+            devLogger.w(RumEventConsumer.RUM_NOT_INITIALIZED_WARNING_MESSAGE)
+            sdkLogger.e(RumEventConsumer.RUM_NOT_INITIALIZED_ERROR_MESSAGE)
+            null
+        } else {
+            rumContext
+        }
+    }
+
+    companion object {
+        const val RUM_NOT_INITIALIZED_WARNING_MESSAGE = "You are trying to use the WebView " +
+            "tracking API but the RUM feature was not properly initialized."
+        const val RUM_NOT_INITIALIZED_ERROR_MESSAGE = "Trying to consume a bundled rum event" +
+            " but the RUM feature was not yet initialized. Missing the RUM context."
```

Comment:
How does this class knows that it is used for webview bundling? Appart from it being in the `webview` package, it could still be generic.

Reply:
I see your point, I am going to try to change its name.